primap2/tests/test_data_reading.py

```diff
@@ -122,6 +122,21 @@ def test_create_na_replacement_dict(strs, user_na_conv, expected_result):
     )
 
 
+@pytest.mark.parametrize(
+    "to_test_entity, to_test_units, to_test_gwp, expected_result",
+    [
+        ("CO2", ["kt CO2 / yr", "mg CO2 / s"], None, "Gg CO2 / yr"),
+        ("CH4", ["kt CO2 / yr"], "AR4GWP100", "Gg CH4 / yr"),
+        ("CH4", ["kt CO2 / yr", "Mg CO2 / yr"], None, "Gg CO2 / yr"),
+    ],
+)
+def test_preferred_unit(to_test_entity, to_test_units, to_test_gwp, expected_result):
+    assert (
+        pm2io._data_reading.preferred_unit(to_test_entity, to_test_units, to_test_gwp)
+        == expected_result
+    )
+
+
```

Comment:
Don't duplicate doctests. Those are already as testable (and tested) examples in the docstring, duplicating them here will just add more work down the line.

Reply:
Sorry, I overlooked the doctests.